include/aspect/initial_conditions/interface.h

```diff
@@ -93,6 +94,31 @@ namespace aspect
     };
 
 
+    /**
+     * Create a class inheriting from dealii.Function to translate the
+     * interface call to dealii.Function format
+     */
+    template <int dim>
+    class InterfaceFunction : public dealii::Function<dim, double>
+    {
+      public:
+        InterfaceFunction (const Interface<dim> &interface,
+                           const unsigned int selected_component,
+                           const unsigned int n_components);
```

Comment:
I don't understand why you need this class. Is this not exactly what `dealii::VectorFunctionFromScalarFunctionObject` does if you present to it a suitably `std::bind`ed (`std::bound`?) object?